lib/membrane/video_compositor/handler/callback_ctx.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Membrane.VideoCompositor.Handler.CallbackCtx do
+  @moduledoc """
+  Structure representing a common part of the context
+  for all of the callbacks.
+  """
+  alias Membrane.{Pad, Time}
+  alias Membrane.VideoCompositor.Scene
+
+  defmacro __using__(fields) do
+    default_fields =
+      quote do
+        [
+          input_pads: list(Pad.ref_t()),
+          scenes_queue: list({scene :: Scene.t(), start_scene_timestamp :: Time.t()}),
+          current_scene: Scene.t()
+        ]
+      end
+
+    quote do
+      @type t :: %__MODULE__{unquote_splicing(fields ++ default_fields)}
+
+      fields_names = unquote(Keyword.keys(fields))
+      default_fields_names = unquote(Keyword.keys(default_fields))
+
+      @enforce_keys Module.get_attribute(__MODULE__, :enforce_keys, fields_names) ++
+                      default_fields_names
+
+      defstruct fields_names ++ default_fields_names
+    end
+  end
```

Comment:
With this macro docs lack `@typedoc`, which IMO would be useful for VC users encountering those `callbacks`. Repeating this logic in every `CallbackCtx.sth` is certainly bad, but I'm curious whether we can somehow generate them, with common parts described here and callback-specific ones (like `earliest_start`) in  `CallbackCtx.sth`.
`default_fields` are rather self-explanatory, but concerted whether it always the case with custom ones.

@DominikWolek What do you think?

Reply:
IMO we can simply write moduledocs for callback contexts

Reply:
With the use of macros we can append the "common" part of the moduledoc to the user-defined moduledoc. 
In Core we do it i.e. here: https://github.com/membraneframework/membrane_core/blob/master/lib/membrane/core/docs_helper.ex
I believe we can also implement that strategy in this case.